src/compiler/utilities.ts

```diff
@@ -1218,7 +1218,23 @@ namespace ts {
     export function isExternalModuleNameRelative(moduleName: string): boolean {
         // TypeScript 1.0 spec (April 2014): 11.2.1
         // An external module name is "relative" if the first term is "." or "..".
-        return moduleName.substr(0, 2) === "./" || moduleName.substr(0, 3) === "../" || moduleName.substr(0, 2) === ".\\" || moduleName.substr(0, 3) === "..\\";

```

Comment:
why not use ` /^\.\.?($|[\\/])/.test(moduleName)`? It's a regex, but I think it's still easier to read than 16 lines of imperative code.